source/builtin_macros/src/unerased_proxies.rs

```diff
@@ -379,6 +379,14 @@ impl crate::syntax::Visitor {
                     &format!("{}{}", VERUS_UNERASED_PROXY, &item_fn.sig.ident),
                     item_fn.sig.span(),
                 );
+                // Remove rustc_diagnostic_item attribute to avoid duplicate diagnostic item errors
+                item_fn.attrs.retain(|attr| {
+                    if let Some(last_seg) = attr.path().segments.last() {
```

Comment:
can you factor this out into 'is_rustc_diagnostic_item_attribute' function or similar?

And please check that `path().segments` has length 1, as well.

Reply:
If someone runs into other unduplicable attributes, we can generalize the solution later. I'm inclined to have this ad-hoc solution now because if it turns out to be sufficient, it has the advantage of not requiring any more documentation. It just works without the developer needing to know about a feature like `#[verifier::no_dup_attribute]`.